File: angular-java-springboot/server/src/main/java/com/lcaohoanq/pfolio/controllers/ProfileController.java
package com.lcaohoanq.pfolio.controllers;

import com.lcaohoanq.pfolio.models.Profile;
import com.lcaohoanq.pfolio.services.IProfileService;
import lombok.RequiredArgsConstructor;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

@RestController
@RequestMapping("/profile")
@RequiredArgsConstructor
public class ProfileController {

    private final IProfileService profileService;

    @GetMapping("")
    public ResponseEntity<Profile> getProfile() {
        return ResponseEntity.ok(profileService.getProfile());
    }

}

File: kotlin-springboot-thymeleaf-rest/src/main/kotlin/com/lcaohoanq/pfolio/demo/domain/profile/ProfileController.kt
package com.lcaohoanq.pfolio.demo.domain.profile

import org.springframework.http.ResponseEntity
import org.springframework.stereotype.Controller
import org.springframework.ui.Model
import org.springframework.web.bind.annotation.GetMapping

@Controller
class ProfileController(
    private val profileService: ProfileService
) {

    @GetMapping("/")
    fun home(model: Model): String {
        model.addAttribute("profile", profileService.getProfile())
        return "home"
    }

    @GetMapping("/api/data")
    fun getMyProfile(): ResponseEntity<Any> {
        return ResponseEntity.ok().body(
            profileService.getProfile()
        )
    }

}